optin/reporting/test/models/completed.spec.js

```diff
@@ -0,0 +1,204 @@
+const runDBNotFoundTests = require('./runDBNotFoundTests');
+const runTotalRecordTests = require('./runTotalRecordTests');
+const Config = require('../../config');
+const completedDB = require('../../models/completed');
+const { refreshDBs } = require('../../models/database');
+const { expect } = require('chai');
+const { stub, restore } = require('sinon');
+const path = require('path');
+
+const runBasicOptinTypeTests = (types, parameters) => {
+	/* eslint-disable no-await-in-loop */
+	for (let i = 0; i <= types.length - 1; ++i) {
+		// Arrange:
+		const optinType = types[i];
+
+		it(`returns ${0 === optinType ? 'pre' : 'post'}-optin records`, async () => {
+			// Act:
+			const result = await completedDB.getCompletedPagination({
+				...parameters,
+				optinType
+			});
+
+			// Assert:
+			result.forEach(item => {
```

Comment:
this is also weak assert, since you're only validating one of many item properties